dags/common/pull_ftp.py

```diff
@@ -66,10 +71,10 @@ def migrate_from_ftp(
     )
 
     if force_pull_all_files:
-        return _force_pull(s_ftp, repo, logger, **kwargs)
+        return _force_pull(s_ftp, repo, logger, publisher, **kwargs)
     elif force_pull_specific_files:
-        return _filenames_pull(s_ftp, repo, logger, **kwargs)
-    return _differential_pull(s_ftp, repo, logger, **kwargs)
+        return _filenames_pull(s_ftp, repo, logger, publisher, **kwargs)
+    return _differential_pull(s_ftp, repo, logger, publisher, **kwargs)
```

Comment:
why having a flag for that? Can we just use a different function for elsevier?

Reply:
we have a flag for it because we have different files migration flow for Elsevier.
The solution with a flag helps to avoid boilerplate code.

Files list comming from  _differential_pull, _filenames_pull or _force_pull are the same to migrate_files or migrate_files_elsevier:
https://github.com/cern-sis/workflows/blob/elsevier-dataset-parsing/dags/common/pull_ftp.py#L97-L99

However, the function of migrate_files and migrate_files_elsevier is different.
Migrate_files_elsevier saves just archives: https://github.com/cern-sis/workflows/blob/elsevier-dataset-parsing/dags/common/pull_ftp.py#L181-L209
While common Migrate_files saves and extracts archives:
https://github.com/cern-sis/workflows/blob/elsevier-dataset-parsing/dags/common/pull_ftp.py#L18-L51

For Elsevier publisher sends us too many files which we don't need. Extracting them takes a lot of time, so we want to extract just what we need.
Elsevier adds mapping in dataset.xml, from where we are figuring out which files we actually need to extract and parse.
The extracting only needed files for Elsevier is happening on trigger function, which triggers files parsing DAG:
https://github.com/cern-sis/workflows/blob/elsevier-dataset-parsing/dags/elsevier/trigger_file_processing.py

For all other publishers don't have mappings, so we are extracting everything that is sent.

Reply:
I see, but `migrate_files_elsevier` and `migrate_files` are very similar, why don't just push the change to this function instead of propagate it in others?

Reply:
there are similar but not the same. 
migrate_files impacts ALL the rest of the publishers, so in the end we have to make more changes than should be the scope of this PR. We can create the task and move it the changes for migrate_files in the other PR. What do you think?

Reply:
the difference between  migrate_files_elsevier and migrate_files is that: migrate_files_elsevier does not extract files, while migrate_files does.
So maybe we can pass the flag extract=False for Elsevier, and then add functionality of migrate_files_elsevier to migrate_files?
for example:
```
def migrate_files(
    archives_names,
    s_ftp: SFTP_FTP_TYPE,
    repo: IRepository,
    logger: PrintLogger,
    extract: bool = False,
):
    logger.msg("Processing files.", filenames=archives_names)
    extracted_or_downloaded_filenames = []
    for archive_name in archives_names:
        logger.msg("Getting file from SFTP.", file=archive_name)
        file_bytes = s_ftp.get_file(archive_name)

        if (
            ".zip" in archive_name
            and zipfile.is_zipfile(file_bytes)
            or ".tar" in archive_name
            and tarfile.is_tarfile(file_bytes)
        ):
            if extract:
                for (archive_file_content, s3_filename) in process_archive(
                    file_bytes=file_bytes, file_name=archive_name
                ):
                    repo.save(s3_filename, io.BytesIO(archive_file_content))
                    if repo.is_meta(s3_filename):
                        extracted_or_downloaded_filenames.append("extracted/" + s3_filename)
                repo.save(archive_name, file_bytes)
            else:
                extracted_or_downloaded_filenames.append("raw" + archive_name)
                repo.save(archive_name, file_bytes)

        else:
            logger.info(
                "File is not zip or tar, processing the next one",
                file_name=archive_name,
            )
            continue
    return extracted_or_downloaded_filenames

```

Reply:
we can rename instad of `extract` to `process_archives` and by default will be `True`